source/s3_client.c

```diff
@@ -1883,12 +1886,26 @@ static void s_s3_client_body_streaming_task(struct aws_task *task, void *arg, en
     while (!aws_linked_list_empty(&payload->requests)) {
         struct aws_linked_list_node *request_node = aws_linked_list_pop_front(&payload->requests);
         struct aws_s3_request *request = AWS_CONTAINER_OF(request_node, struct aws_s3_request, node);
+        struct aws_s3_meta_request *meta_request = request->meta_request;
+
+        aws_s3_meta_request_lock_synced_data(meta_request);
```

Comment:
I don't think it's worth to be a function? Since, the lock we are acquiring. The check can happen inside a lock block, and a function will make it a little bit awkward.

Reply:
In this case, I think moving the lock and state check to a small function would "encapsulate" it a bit.  Doing checks on the state inside of the meta request are totally fine, but because we've left that area of code, it might be kind of nice to bundle this up so other places could also poll if a meta request is active.